rpc/connection.go

```diff
@@ -609,6 +609,9 @@ func (c *Connection) doReconnect(ctx context.Context, disconnectStatus Disconnec
 			// A fatal error happened.
 			*reconnectErrPtr = err
 			// short-circuit Retry
+			if err != nil {
+				c.transport.Close()
```

Comment:
Existing code (by that I mean the mdserver/bserver clients) doesn't call `Dial` at all.  It just uses the existing `Connection` object until a ping is lost or it gets a connectivity check request.  From what I understand, the way things work now, a new RPC function call will end up kicking off reconnect again, right?  I'm not sure what happens if you try reconnect on a connection that has a closed transport.